Mathlib/LinearAlgebra/AffineSpace/Independent.lean

```diff
@@ -839,21 +839,31 @@ theorem range_face_points {n : ℕ} (s : Simplex k P n) {fs : Finset (Fin (n + 1
 
 /-- The face of a simplex with all but one point. -/
 def faceOpposite {n : ℕ} [NeZero n] (s : Simplex k P n) (i : Fin (n + 1)) : Simplex k P (n - 1) :=
-  s.face (fs := {j | j ≠ i}) (by simp [filter_ne', NeZero.one_le])
+  s.face (fs := {i}ᶜ) (by simp [card_compl, NeZero.one_le])
 
 @[simp] lemma range_faceOpposite_points {n : ℕ} [NeZero n] (s : Simplex k P n) (i : Fin (n + 1)) :
-    Set.range (s.faceOpposite i).points = s.points '' {j | j ≠ i} := by
+    Set.range (s.faceOpposite i).points = s.points '' {i}ᶜ  := by
   simp [faceOpposite]
 
+/-- Needed to make the affine span nonempty. -/
+instance {n : ℕ} [NeZero n] (i : Fin (n + 1)) : Nonempty ({i}ᶜ : Set _) :=
+  .intro ⟨i + 1, by simp [NeZero.ne]⟩
```

Comment:
This feels like it should be two instances, neither of them in this file (don't know if either already exists). One would deduce `Nontrivial (Fin (n + 1))` from `[NeZero n]`, and the other would make the `Nonempty` deduction for an arbitrary `Nontrivial` type.

Reply:
Both of those almost already existed, hopefully 1faeeca04a2ed58b72545a630b3a2ed0f08b0e40 is what you were thinking of.

Reply:
My argument would be that we might only want the instance in the `AffineSpace` scope; in general we probably don't want to duplicate the entire `Set.Nonempty` APi without further discussion.